### PR TITLE
requirements: update craft-parts to 1.17.1

### PR DESCRIPTION
**Pull request by cmatsuoka**
This enables git+ssh:// scheme in in git sources.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `make lint`?
- [ ] Have you successfully run `pytest tests/unit`?

-----

### CODE REVIEW

File: requirements-devel.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ codespell==2.2.1
 coverage==6.5.0
 craft-cli==1.2.0
 craft-grammar==1.1.1
-craft-parts==1.15.0
+craft-parts==1.17.1
```
Comment (sergiusens) [inline requirements-devel.txt:14]: this is 2 versions of bumping for hotfix, what else are we taking in that could be risky?

Comment (cmatsuoka) [reply requirements-devel.txt:14]: There are many changes there, we can backport the git scheme change to 1.15 series and use that instead.

Comment (cmatsuoka) [reply requirements-devel.txt:14]: PR #3993 created to update the 7.2 hotfix branch to craft-parts 1.15.2. This PR retargeted to main to update it with the corresponding change.

**Comment by cmatsuoka** [comment]
The craft-parts update to 1.17.1 now targets the main branch.

**Pull request merged**